sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -109,7 +172,19 @@ public static BinaryData Write(object model, ModelReaderWriterOptions? options =
 
             options ??= ModelReaderWriterOptions.Json;
 
-            return GetInstance<T>().Create(data, options);
+            Type typeOfT = typeof(T);
+
+            if (typeOfT.IsGenericType && typeOfT.GetGenericTypeDefinition().Equals(typeof(List<>)))
```

Comment:
Would we only support `List<T>`?  What about array, read-only list, or other collection type?